### PR TITLE
#167164984 Enable social login for Facebook and Google

### PR DESCRIPTION
**Pull request by Rythae**
#### Pivotal Tracker Story

[#167164984](https://www.pivotaltracker.com/story/show/167164984)

#### What does this PR do?
Setup oauth for social login

#### Summary of Task
 - setup passport configuration
 - setup passport strategies
- write test for oauth routes
 
#### How can this be tested?
- git clone branch `feature/167164984/enable-social-login`
- install passport, passport-facebook, passport-google-oauth
- install dependencies
- run npm install
- start dev server npm run dev:start

#### Screenshots (if appropriate)
None

### CODE REVIEW

**Comment by Ukhu** [comment]
Kindly address HoundCI issues

File: src/routes/index.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import express from 'express';
 import user from './user';
 import auth from './auth';
+import oauth from './oauth';
 
 const router = express.Router();
 
-router.use('/', user, auth);
+// router.use('/oauth', oauth);
```
Comment (OvieMudi) [inline src/routes/index.js]: Hey, remove the commented code.

File: tests/passport.spec.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import {
```
[the remark this answers is not in the document]
Comment (OvieMudi) [reply tests/passport.spec.js:1]: Please see to this

**Comment by profmcdan** [comment]
@Rythae There is a merge conflict and a code coverage issue on this PR. Kindly take a look at it

File: tests/passport.spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rythae) [reply tests/passport.spec.js:1]: Alright

**Comment by Rythae** [comment]
This feature will be completed in the frontend

**Review by Nkemjiks — APPROVED** [review]

**Pull request merged**